x-pack/plugins/fleet/server/services/epm/archive/validation.ts

```diff
@@ -189,107 +204,133 @@ export function parseAndVerifyDataStreams(
     const {
       title: dataStreamTitle,
       release,
-      ingest_pipeline: ingestPipeline,
       type,
       dataset,
+      ingest_pipeline: ingestPipeline,
+      streams: manifestStreams,
+      ...restOfProps
     } = manifest;
     if (!(dataStreamTitle && release && type)) {
       throw new PackageInvalidArchiveError(
         `Invalid manifest for data stream '${dataStreamPath}': one or more fields missing of 'title', 'release', 'type'`
       );
     }
-    const streams = parseAndVerifyStreams(manifest, dataStreamPath);
+    const streams = parseAndVerifyStreams(manifestStreams, dataStreamPath);
 
     // default ingest pipeline name see https://github.com/elastic/package-registry/blob/master/util/dataset.go#L26
-    return dataStreams.push({
-      dataset: dataset || `${pkgName}.${dataStreamPath}`,
-      title: dataStreamTitle,
-      release,
-      package: pkgName,
-      ingest_pipeline: ingestPipeline || 'default',
-      path: dataStreamPath,
-      type,
-      streams,
-    });
+    dataStreams.push(
+      Object.entries(restOfProps).reduce(
+        (validatedDataStream, [key, value]) => {
+          if (registryDataStreamProps.includes(key as RegistryDataStreamKeys)) {
+            // @ts-ignore-error
+            validatedDataStream[key] = value;
+          }
+          return validatedDataStream;
+        },
+        {
+          title: dataStreamTitle,
+          release,
+          type,
+          package: pkgName,
+          dataset: dataset || `${pkgName}.${dataStreamPath}`,
+          ingest_pipeline: ingestPipeline || 'default',
+          path: dataStreamPath,
+          streams,
+        } as RegistryDataStream
+      )
```

Comment:
Not sure if it helps, but https://github.com/elastic/kibana/pull/85711/commits/3f213e5bb634adc5688a8ea539578717efe9ef4e has some code which has to do similar validation in a type-safe way

Reply:
I saw this code when I worked on this PR, but it required additional declaration of `requiredArchivePackageProps` and `optionalArchivePackageProps`, which are not automatically kept in sync with `ArchivePackage`. I wanted to avoid the duplication